presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcClient.java

```diff
@@ -36,6 +37,8 @@
 
     List<JdbcColumnHandle> getColumns(JdbcTableHandle tableHandle);
 
+    Optional<ReadMapping> toPrestoType(JdbcTypeHandle typeHandle);
```

Comment:
Adding this to API is necessary to remove type conversion logic from `JdbcRecordCursor`. 
( BTW I decided to use `JdbcTypeHandle` (a new type) as a parameter, instead of `int jdbcType, int columnSize, int decimalDigits` we had, for the sake of future API stability.)

@electrum i hope adding this method doesn't break anything on your side.